ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -247,25 +248,33 @@ private KeyField findKeyField(final List<SelectExpression> selectExpressions) {
         return Optional.empty();
       }
 
-      final Field keyField = new Field(
-          getKeyField().name().get(), -1, Schema.OPTIONAL_STRING_SCHEMA);
+      final String fullFieldName = getKeyField().name().get();
+      final Optional<String> fieldAlias = SchemaUtil.getFieldNameAlias(fullFieldName);
+      final String fieldNameWithNoAlias = SchemaUtil.getFieldNameWithNoAlias(fullFieldName);
+
+      final Field keyField = fieldAlias
```

Comment:
I've seen this pattern a few times - I think maybe we could benefit from a `Field.of(Optional<String>, String, SqlType` or maybe a `.withAlias(Optional<String>)` so that we can just do `keyField = Field.of(fieldNameWithNoAlias, STRING).withAlias(fieldAlias)`

Reply:
Done.